src/Passes/docs/src/UserGuide/BuildingLibrary.md

```diff
@@ -0,0 +1,149 @@
+
+## Prerequisites
+
+It is necessary for you to have the following tools before you get started on making a new build from this project:
+
+- C++ compiler
+- CMake
+
+If you are compiling and using the command-line tool, these are necessary tools regardless of whether you plan to develop the library itself. The build configuration is handled by CMake, which informs you if a compiler needs to be installed.
+
+### On OS X
+
+Those using OS X can install CMake through the use of a command-line tool called `brew`, as follows:
+
+```sh
+brew install cmake
+```
+
+It should not even be necessary to install anything else on top of OS X, since the compiler comes preinstalled. There is no need to read the next subsection ([Developer prerequisites](#developer-prerequisites)) if you are not intending to modify the core library and can safely jump to the section [Library Dependencies](#library-dependencies)
+
+### On Ubuntu 20.04
+
+Additionally to CMake, you will also require Clang in order to build Ubuntu 20.04. You can accomplish this by following the steps below:
+
+```sh
+apt install clang-11 cmake
+```
+
+You can skip straight to [Library Dependencies](#library-dependencies) if you do not intend to work on developing the core library.
+
+## Developer prerequisites
+
+Development of the core library requires additional tools. These are used formatting, linting and managing code quality. For this purpose, the following tools and packages are used:
+
+- Python 3
+- Python packages specified in `requirements.txt`
+- clang-format
+- clang-tidy
+
+The installation process varies depending on the platform you use. The following subsections provide details on how to install these tools on each platform.
+
+### On OS X
+
+TODO:
+
+### On Ubuntu 20.04
+
+Installing the clang tools on Ubuntu along with Python can be accomplished by running these commands:
+
+```sh
+apt install clang-format-11 clang-tidy-11
+apt install python3 python3-pip
+```
+
+We recommend that you use version 11 of clang in order to be consistent with the version of LLVM on which the library depends. In general, the code should work with any version of clang. Make sure that the compiler environment variables are defined correctly in order to select the correct version:
+
+```sh
+export CC=clang-11
+export CXX=clang++-11
+```
+
+### Common to all platforms
+
+Last but not least, we install the Python libraries that are required:
+
+```sh
+      pip install -r requirements.txt
+      chmod +x manage
```

Comment:
`chmod` isn't used on Windows, but this step is listed as being "common to all platforms."

Reply:
This would need to be revised once there is support for Windows. There are a number of places where the current setup is unlikely to work on this platform as it stands.